### DISCUSSION TITLE
Grid layout system / flex or container

### DISCUSSION THREAD

**Discussion by aleczratiu** [tailwindlabs/tailwindcss#12423] [Help]
Is there anyone who tried any of these approaches to have a standard system for the layout?
What is your approach, are you using the 12 - 8 - 4 grid system or '@component' library or flex approach?

I have an example of a figma with grid design system, can be done something like this?
![Screenshot 2023-11-15 at 10 05 24 PM](https://github.com/tailwindlabs/tailwindcss/assets/33958032/3e928982-6a09-4bbf-a50f-3ae27b8189df)

Or should I define breakpoints with the rems like in grid system? I was looking on multiple websites which use tailwind and to be honest everywhere's applied a mix and match between flex and grid, or only flex. What is your suggestion and experience with it?

**Comment by alex-krasikau** [comment]
Hey @aleczratiu!

Most often, I use 12 grid all the way, so I add the following class to my container components:

```
<div class="grid grid-cols-12 gap-x-6">
  <div class="col-span-1 lg:col-span-4 xl:col-span-2">...</div>
</div> 
```

But sometimes, I use other variants if the end solution seems to be more straightforward:
```
<div class="grid grid-cols-1 gap-x-6 lg:grid-cols-2 xl:grid-cols-3">
  <div>...</div>
</div>
```

**Comment by aleczratiu** [comment]
Thank you @alex-krasikau for your reply!
I've started using the flex with the rules from grid, which means all the grids from the images match the tailwind breakpoints without the left and right gutters.
So I moved the left-right gutters to padding and now I use max-w-[82] (for 2xl screens) and px-8 which represents the 2 gutter for 2xl screen (left-right), in this way I use the flex with the rules of 12-8-4 columns. Not sure if is the best approach to use like this but is working decent.
